app/frontend/src/lib/currentLocation.test.js

```diff
@@ -0,0 +1,56 @@
+import currentLocation, { onSuccess, onFailiure, showErrorMessage } from './currentLocation';
+import radius from '../search/ui/input/radius';
+
+jest.mock('../search/ui/input/radius');
+
+describe('location search box', () => {
+  let showErrorMessageMock = null; let stopLoadingMock = null; let enableRadiusMock = null; let
+    disableRadiusMock = null;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    currentLocation.showErrorMessage = jest.fn();
+    showErrorMessageMock = jest.spyOn(currentLocation, 'showErrorMessage');
+
+    currentLocation.stopLoading = jest.fn();
+    stopLoadingMock = jest.spyOn(currentLocation, 'stopLoading');
+
+    enableRadiusMock = jest.spyOn(radius, 'enableRadiusSelect');
+    disableRadiusMock = jest.spyOn(radius, 'disableRadiusSelect');
+
+    document.body.innerHTML = '<div class="js-location-finder"><input type="text" id="location" class="js-location-finder__input" /></div>';
+  });
+
+  describe('onFaliure', () => {
+    test('returns true if location is in predefined list', () => {
```

Comment:
Are these the correct test names?

Reply:
ah well spotted, theres probable describe block copy and pasted all over this file :D